### ISSUE TITLE
Curl setopt - error: call to throwing function ioerror without throws, try, or try!

### ISSUE THREAD

**Issue by ljdursi** [chapel-lang/chapel#7570]
Hi, all:

### Summary of Problem

In trying to use the setopt function with curl, at compilation time a compile error is thrown about setopt calling ioerror without a throws/try/try!:

```
$ chpl -o testcurl testcurl.chpl
$CHPL_HOME/modules/packages/Curl.chpl:302: In function 'setopt':
$CHPL_HOME/modules/packages/Curl.chpl:313: error: call to throwing function ioerror without throws, try, or try! (relaxed mode)
$CHPL_HOME/modules/standard/SysError.chpl:407: note: throwing function ioerror defined here
```

### Steps to Reproduce

**Source Code:**

```chapel
use Curl;

proc main() {
  const uniprotid = "P07204_TRBM_HUMAN";
  const uniproturl = "http://www.uniprot.org/uniprot/" + uniprotid + ".fasta";
  var response = "";

  try! {
    var c = open(url=uniproturl, mode=iomode.r);
    c.setopt(curlopt_followlocation, 1);
    var reader = c.reader();
    var line: string;

    while (reader.readline(line)) do
      response += line;
  }

  writeln(uniprotid);
  writeln(response);
}
```

**Compile command:**

`chpl -o testcurl testcurl.chpl`

### Fix?

A crude change to get setopt to compile seems to work:

```
302c302
< proc file.setopt(opt:c_int, arg):bool {
---
> proc file.setopt(opt:c_int, arg):bool throws {
313c313,315
<   if err then ioerror(err, "in file.setopt(opt:c_int, arg)");
---
>   try! {
>     if err then ioerror(err, "in file.setopt(opt:c_int, arg)");
>   }
746c748
< } /* end of module */
\ No newline at end of file
---
> } /* end of module */
```

### Configuration Information

- Output of `chpl --version`:

```
chpl Version 1.16.0
Copyright (c) 2004-2017, Cray Inc.  (See LICENSE file for more details)
```

- Output of `$CHPL_HOME/util/printchplenv --anonymize`:

```
CHPL_TARGET_PLATFORM: darwin
CHPL_TARGET_COMPILER: clang
CHPL_TARGET_ARCH: native
CHPL_LOCALE_MODEL: flat
CHPL_COMM: none
CHPL_TASKS: qthreads
CHPL_LAUNCHER: none
CHPL_TIMERS: generic
CHPL_UNWIND: none
CHPL_MEM: jemalloc
CHPL_MAKE: make
CHPL_ATOMICS: intrinsics
CHPL_GMP: gmp
CHPL_HWLOC: hwloc
CHPL_REGEXP: re2
CHPL_WIDE_POINTERS: struct
CHPL_AUX_FILESYS: curl *
```

- Back-end compiler and version, e.g. `gcc --version` or `clang --version`:

```
$ clang --version
Apple LLVM version 9.0.0 (clang-900.0.37)
Target: x86_64-apple-darwin17.0.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

**Comment by bradcray** [comment]
@psahabu and @mppf: could one of you take a look?

**Comment by psahabu** [comment]
I'm working on a fix, sorry about that oversight.

**Comment by psahabu** [comment]
Sorry for the delay @ljdursi. The fix has been merged in #7713, and testing for the Curl module is on its way.

**Comment by ljdursi** [comment]
Thanks!
